src/code/z_play.c

```diff
@@ -1,59 +1,1464 @@
+                    }
+                }
+
+                if ((HREG(80) != 10) || (HREG(93) != 0)) {
+                    DebugDisplay_DrawObjects(globalCtx);
+                }
+
+                if ((R_PAUSE_MENU_MODE == 1) || (D_80161490 == 1)) {
+                    Gfx* sp70 = gfxCtx->overlay.p;
+                    globalCtx->preRenderCtx.unk_10 = gfxCtx->curFrameBuffer;
+                    globalCtx->preRenderCtx.unk_14 = gZBuffer;
+                    func_800C1F20(&globalCtx->preRenderCtx, &sp70);
+                    if (R_PAUSE_MENU_MODE == 1) {
+                        globalCtx->preRenderCtx.unk_18 = gfxCtx->curFrameBuffer;
+                        func_800C20B4(&globalCtx->preRenderCtx, &sp70);
+                        R_PAUSE_MENU_MODE = 2;
+                    } else {
+                        D_80161490 = 2;
+                    }
+                    gfxCtx->overlay.p = sp70;
+                    globalCtx->preRenderCtx.unk_A3 = 2;
+                    SREG(33) |= 1;
+                } else {
+                Gameplay_Draw_DrawOverlayElements:
+                    if ((HREG(80) != 10) || (HREG(89) != 0)) {
+                        Gameplay_DrawOverlayElements(globalCtx);
+                    }
+                }
+            }
+        }
+    }
+
+    if (globalCtx->view.unk_124 != 0) {
+        Vec3s sp50;
+        func_800591EC(&sp50, globalCtx->cameraPtrs[globalCtx->activeCamera]);
+        func_800AB944(&globalCtx->view);
+        globalCtx->view.unk_124 = 0;
+        if ((globalCtx->skyboxId != 0) && (globalCtx->skyboxId != 0x1D) && !globalCtx->envCtx.skyDisabled) {
+            func_800B1030(&globalCtx->skyboxCtx, globalCtx->view.eye.x, globalCtx->view.eye.y, globalCtx->view.eye.z);
+        }
+    }
+
+    func_80059EC8(globalCtx->cameraPtrs[globalCtx->activeCamera]);
+
+    Graph_CloseDisps(dispRefs, gfxCtx, "../z_play.c", 4508);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Gameplay_Draw.s")
+#endif
+
+void Gameplay_Main(GlobalContext* globalCtx) {
+    D_8012D1F8 = &globalCtx->state.input[0];
+
+    DebugDisplay_Init();
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 4556);
+    }
+
+    if ((HREG(80) == 10) && (HREG(94) != 10)) {
+        HREG(81) = 1;
+        HREG(82) = 1;
+        HREG(83) = 1;
+        HREG(84) = 3;
+        HREG(85) = 1;
+        HREG(86) = 1;
+        HREG(87) = 1;
+        HREG(88) = 1;
+        HREG(89) = 1;
+        HREG(90) = 15;
+        HREG(91) = 1;
+        HREG(92) = 1;
+        HREG(93) = 1;
+        HREG(94) = 10;
+    }
+
+    if ((HREG(80) != 10) || (HREG(81) != 0)) {
+        Gameplay_Update(globalCtx);
+    }
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 4583)
+    }
+
+    Gameplay_Draw(globalCtx);
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 4587);
+    }
+}
+
+// original name: "Game_play_demo_mode_check"
 s32 func_800BFC84(GlobalContext* globalCtx) {
```

Comment:
Same comment as above, since we know the name why not fit a name to our scheme.
```suggestion
s32 Gameplay_DemoModeCheck(GlobalContext* globalCtx) {
```